Lib/socketserver.py

```diff
@@ -651,24 +652,32 @@ def process_request_thread(self, request, client_address):
         except Exception:
             self.handle_error(request, client_address)
         finally:
-            self.shutdown_request(request)
+            try:
+                self.shutdown_request(request)
+            finally:
+                thread = threading.current_thread()
+                with self._threads_lock:
+                    if self._threads and not thread.daemon:
```

Comment:
I would prefer to test: "self._threads is not None".

Would you mind to test "if not thread.daemon" before "with self._threads_lock:"?

```
if not thread.daemon:
   with self._threads_lock:
     if self._threads  is not None: ...
```

It's just to avoid holding the lock if it's useless.

Reply:
In the latest commit, `_threads.remove_current()` does not hold the lock for daemon threads.